go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -58,40 +62,42 @@ func (agent *ActionAgent) Backup(ctx context.Context, concurrency int, logger lo
 	}
 	originalType := tablet.Type
 
-	// update our type to BACKUP
-	if _, err := topotools.ChangeType(ctx, agent.TopoServer, tablet.Alias, topodatapb.TabletType_BACKUP); err != nil {
-		return err
-	}
+	if *mysqlctl.BackupEngineImplementation != xtrabackupEngine {
```

Comment:
I wonder if there is a way we could push down all this logic to the engine itself. It seems like in the future, adding an agent requires to be aware of this context too.

Reply:
That would have been ideal but I didn't see a way to achieve it. I'm happy to fix it if I'm wrong about that and someone can point me in the right direction.

Reply:
Yeah. I couldn't see a clear path either. I wonder if @sougou  has some thoughts here.

Reply:
Changed this to use GetBackupEngine instead of BackupEngineImplementation. That also allows us to not export BackupEngineImplementation.